src/security/ServerOptions.cc

```diff
@@ -196,8 +196,9 @@ Security::ServerOptions::initServerContexts(AnyP::PortCfg &port)
 
     if (generateHostCertificates) {
         createSigningContexts(port);
+    }
 
-    } else if (!createStaticServerContext(port)) {
+    if (!certs.empty() && !createStaticServerContext(port)) {
```

Comment:
Is it possible that generateHostCertificates is false and certs are empty? If it is possible, then please add a source code comment (or adjust the code) to clarify what happens in that case. If it is impossible, then please replace the `!certs.empty()` part of the condition with an assert.

Reply:
It is possible when generate-host-certificates=off is configured and tls-cert= is omitted.

That is one of the correct situations where one of these errors is expected to be shown:
 FATAL: No valid signing certificate configured for HTTP_port 0.0.0.0:3128
 ERROR: Ignoring http_port [::]:8082 due to TLS context initialization failure.

(which one depends on other details)

Reply:
If it is possible, then please add a source code comment (or adjust the code) to clarify what happens in that case.